src/deluge/memory/cache_manager.cpp

```diff
@@ -0,0 +1,176 @@
+					Debug::print(" to ");
+					Debug::println(appropriateQueue);
+
+					auto* next = static_cast<Stealable*>(reclamation_queue_[q].getNext(stealable));
+
+					stealable->remove();
+					this->QueueForReclamation(appropriateQueue, stealable);
+
+					stealable = next;
+					continue;
+				}
+			}
+
+			// Ok, we've got one Stealable
+			uint32_t* __restrict__ header = (uint32_t*)((uint32_t)stealable - 4);
+			spaceSize = (*header & SPACE_SIZE_MASK);
+
+			stealable->lastTraversalNo = currentTraversalNo;
+
+			// How much additional space would we need on top of this Stealable?
+			int amountToExtend = totalSizeNeeded - spaceSize;
+
+			newSpaceAddress = (uint32_t)stealable;
+
+			// If that one Stealable alone was big enough, that's great
+			if (amountToExtend <= 0) {
+				found = true;
+				break;
+			}
+
+			// Otherwise, see if available neighbouring memory adds up to make enough in total
+			NeighbouringMemoryGrabAttemptResult result = region.attemptToGrabNeighbouringMemory(
+			    stealable, spaceSize, amountToExtend, amountToExtend, thingNotToStealFrom, currentTraversalNo, true);
+
+			// We also told that function to steal the initial main Stealable we are looking at, once it has ascertained that there is enough memory in total.
+			// Previously I attempted to have it steal everything but that central Stealable, and we would steal that afterwards, down below, but this could go wrong
+			// as thefts occurring in the above call to attemptToGrabNeighbouringMemory() could themselves cause other memory to be deallocated or shortened -
+			// and what if this happened to our main, central Stealable before we actually steal it?
+			// This was certainly a problem in automated testing, though I haven't quite wrapped my head around whether this would quite occur under real operation -
+			// but oh well, there is no harm in taking the safe option.
+
+			// If that couldn't be done (in which case the original, central Stealable won't have been stolen either), move on to next Stealable to assess
+			if (!result.address) {
+				if (result.longestRunFound > longestRunSeenInThisQueue) {
+					longestRunSeenInThisQueue = result.longestRunFound;
+				}
+				stealable = static_cast<Stealable*>(reclamation_queue_[q].getNext(stealable));
+				continue;
+			}
+
+			newSpaceAddress = result.address;
+
+			spaceSize += result.amountsExtended[0] + result.amountsExtended[1];
+
+			Debug::println("stole and grabbed neighbouring stuff too...........");
+			stolen = true;
+			break;
+		}
+
+		// End of that particular queue - so go to the next one
+		longest_runs_[q] = longestRunSeenInThisQueue;
+		currentTraversalNo++;
+	}
+
+	if (!(found || stolen)) {
+#if TEST_GENERAL_MEMORY_ALLOCATION
+		skipConsistencyCheck = false;
+#endif
+		AudioEngine::logAction("/CacheManager::reclaim nope");
+		return 0;
+	}
+
+	if (found && !stolen) {
+		// Warning - for perc cache Cluster, stealing one can cause it to want to allocate more memory for its list of zones
```

Comment:
What does this mean? And why does it send up an error i007?

Reply:
It means that a block of potentially free-able memory has been found, but not yet stolen. Honestly don't know why it triggers that error, but it's a direct translation of the `foundIt` label here:
https://github.com/SynthstromAudible/DelugeFirmware/blob/6c915942ee0383c93f27f2eff8bfd9bc353532c7/src/deluge/memory/memory_region.cpp#L382